Implement generic NxN matrix determinant

sermuns opened the pull request:
Closes #4

PabloRuizCuevas commented:
Nice this is a very cool addition,

I'm thinking for the case of 2x2 we can let the original implementation as it may have less overhead, and maybe put a couple of tests in test.typ, what do you think?

sermuns commented:
Sure, sounds reasonable.

About testing. Have you considered incorporating [tytanic](https://github.com/typst-community/tytanic)? I could try implementing it in another PR, to see if it makes testing better/easier here?

sermuns commented:
Now it fallbacks to simple implementation when matrix is 2x2.

I also added tests for three examples. I noticed we were importing `main.typ` in test file, was that intended? I changed it to `lib.typ`, but was this perhaps wrong?

PabloRuizCuevas commented:
@sermuns , perfect thanks, I will publish the new version with this addition.

Never seen tytanic before, but a test runner with github actions CI, looks promising. So sure, feel free.
